src/main/scala/is/hail/methods/PCRelate.scala

```diff
@@ -29,34 +33,173 @@ object PCRelate {
+    maf: Double,
+    blockSize: Int,
+    minKinship: Double,
+    storageLevel: StorageLevel,
+    statistics: PCRelate.StatisticSubset): Table = {
+
+    val result = new PCRelate(maf, blockSize, statistics, storageLevel)(blockedG, pcs)
+
+    PCRelate.toTable(hc, result, columnKeys, columnKeyType, blockSize, minKinship, statistics)
+  }
+
+  def apply(
+    hc: HailContext,
+    blockedG: M,
+    columnKeys: Array[Annotation],
+    columnKeyType: Type,
+    pcaScores: Array[IndexedSeq[Double]],
+    maf: Double,
+    blockSize: Int,
+    minKinship: Double,
+    storageLevel: StorageLevel,
+    statistics: PCRelate.StatisticSubset): Table = {
+
+    val pcs = rowsToBDM(pcaScores)
+
+    apply(hc, blockedG, columnKeys, columnKeyType, pcs, maf, blockSize, minKinship, storageLevel, statistics)
+  }
+
+  def apply(
+    hc: HailContext,
+    blockedG: M,
+    scoresTable: Table,
+    maf: Double,
+    blockSize: Int,
+    minKinship: Double,
+    statistics: PCRelate.StatisticSubset): Table = {
+    
+    val scoresLocal = scoresTable.collect()
+    assert(scoresLocal.length == blockedG.nCols)
+
+    val keyQueriers = scoresTable.key.map(scoresTable.signature.query(_))        
+    val keys = scoresLocal.map(a => Row.fromSeq(keyQueriers.map(q => q(a)))).toArray[Annotation] 
+    
+    val keyType = scoresTable.keySignature
+
+    val scoresQuerier = scoresTable.signature.query("scores")
+    val pcaScores = scoresLocal.map(a => scoresQuerier(a).asInstanceOf[IndexedSeq[Double]])
+
+    val storageLevel = defaultStorageLevel
+
+    apply(hc,
+      blockedG,
+      keys,
+      keyType,
+      pcaScores,
+      maf,
+      blockSize,
+      minKinship,
+      storageLevel,
+      statistics)
+  }
+
+  def apply(vds: MatrixTable,
+    pcaScores: Array[Array[Double]],
+    maf: Double,
+    blockSize: Int): Table =
+    apply(vds, pcaScores, maf, blockSize, defaultMinKinship, defaultStorageLevel, defaultStatisticSubset)
+
+  def apply(vds: MatrixTable,
+    pcaScores: Array[Array[Double]],
+    maf: Double,
+    blockSize: Int,
+    statisticSubset: StatisticSubset): Table =
+    apply(vds, pcaScores, maf, blockSize, defaultMinKinship, defaultStorageLevel, statisticSubset)
+
+  def apply(vds: MatrixTable,
+    pcaScores: Array[Array[Double]],
+    maf: Double,
+    blockSize: Int,
+    minKinship: Double): Table =
+    apply(vds, pcaScores, maf, blockSize, minKinship, defaultStorageLevel, defaultStatisticSubset)
+
+  def apply(vds: MatrixTable,
+    pcaScores: Array[Array[Double]],
+    maf: Double,
+    blockSize: Int,
+    minKinship: Double,
+    storageLevel: StorageLevel,
+    statistics: PCRelate.StatisticSubset): Table = {
+
+    vds.requireUniqueSamples("pc_relate")
+    val g = vdsToMeanImputedMatrix(vds)
+    val blockedG = BlockMatrix.fromIRM(g, blockSize).cache()
+
+    val pcaScores0: Array[IndexedSeq[Double]] = pcaScores.map(_.toIndexedSeq)
+    
+    apply(vds.hc,
+      blockedG,
+      vds.stringSampleIds.toArray[Annotation],
```

Comment:
Can't we delete every signature that takes an array of annotations and just use integer indices?